salsah/src/public/js/jquery.searchbox.js

```diff
@@ -217,6 +224,15 @@
 								$this.css({cursor: 'text'});
 							}
 							localdata.ele.sel.hide();
+							// when leaving this input, if the value is empty, but we had a previous valid input
+							// set back the previous input
+							if ($this.attr("isValid") && $this.attr("isValid") == "false") {
```

Comment:
not at all, you are completely right as stated on page 109, I will _make [myself] a better programmer_ and avoid `==` from now on.